### PR TITLE
WIP: flit 2.0

### PR DESCRIPTION
**Pull request by takluyver**
Follows discussion on PR #15.

### CODE REVIEW

**Comment by takluyver** [comment]
This is currently failing because pip can't bootstrap flit_core (I'm fixing that in https://github.com/pypa/pip/pull/7394).

Maybe when I'm more awake I'll see a way around that. Otherwise, we'll either have to wait for a release of pip with that fix, or make a conda package of pep517 (which provides a minimal/proof of concept frontend which can bootstrap it).

**Comment by takluyver** [comment]
The important bit of the error now looks to be:

```
INFO conda_build.build:build(1571): Packaging flit
Packaging flit
/home/conda/feedstock_root/build_artifacts/flit_1574762454456/work/install-flit.sh: line 4: /home/conda/feedstock_root/build_artifacts/flit_1574762454456/_h_env_placehold_...placehold_placeh/bin/python: No such file or directory
```

**Comment by ocefpaf** [comment]
Locally this is failing for me with:

```
Fetching list of valid trove classifiers               I-flit.validate
The file description seems not to be valid rst for PyPI; it will be interpreted as plain text  E-flit.validate
<string>:25: (WARNING/2) Cannot analyze code. Pygments package not found.
  E-flit.validate
Invalid config values (see log)
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -33,11 +31,48 @@ requirements:
...
+    script: install-core.bat  # [win]
+    requirements:
+      host:
+        - python
+        - pip
+        - pytoml
+      run:
+        - python
+        - pytoml
+      test:
```
Comment (saraedum) [inline recipe/meta.yaml]: The indent here looks wrong to me. This should not be in `requirements` I think.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,17 +1,15 @@
-{% set version = "1.3" %}
+{% set version = "2.1.0" %}
 
 package:
-  name: flit
+  name: flit-split
```
Comment (saraedum) [inline recipe/meta.yaml]: It's my understanding that this will create a `flit-split` metapackage. Is that what you intended here?

Comment (chrisburr) [reply recipe/meta.yaml]: Not in this case. The name is ignored in this case as it isn't mentioned in `outputs`.

A better way of working around this conda-build "feature" is to use the name of the first output (`flit-core` in this case).

**Comment by chrisburr** [comment]
I'm about to push a tidied version of this PR that I made while testing the fix for https://github.com/conda-forge/flit-feedstock/pull/17#issuecomment-558559492.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
-  sha256: 6f6f0fb83c51ffa3a150fa41b5ac118df9ea4a87c2c06dff4ebf9adbe7b52b36
+  sha256: 6c954c3ec4e87a84fac55cca1a1d692a39354affa7c791eb49f6631b71ddbb40
 
 build:
   number: 0
-  script: FLIT_ROOT_INSTALL=1 {{PYTHON}} -m flit install --deps=none  # [not win]
-  script: {{PYTHON}} -m flit install --deps=none  # [win]
   skip: True  # [py2k]
   entry_points:
     - flit = flit:main
```
Comment (saraedum) [inline recipe/meta.yaml]: I am not sure, but I would assume that the entrypoints also need to go into the corresponding output section.

**Comment by saraedum** [comment]
> Locally this is failing for me with:
> 
> ```
> Fetching list of valid trove classifiers               I-flit.validate
> The file description seems not to be valid rst for PyPI; it will be interpreted as plain text  E-flit.validate
> <string>:25: (WARNING/2) Cannot analyze code. Pygments package not found.
>   E-flit.validate
> Invalid config values (see log)
> ```

Adding pygments as a requirement of the flit output fixes this for me. The buid then works for me with the above changes. It's a bit odd since this says "warning".

**Comment by takluyver** [comment]
The pygments thing is https://github.com/takluyver/flit/issues/297

**Comment by takluyver** [comment]
Thanks everyone for jumping in on this. I'll close this PR in favour of #19.

**Pull request closed without merging**